dockstore-integration-testing/src/test/java/io/dockstore/client/cli/GeneralWorkflowIT.java

```diff
@@ -111,6 +111,93 @@ private Workflow manualRegisterAndPublish(WorkflowsApi workflowsApi, String work
         return workflow;
     }
 
+    /**
+     * This tests that smart refresh correctly refreshes the right versions based on some scenarios for GitHub
+     */
+    @Test
+    public void testSmartRefreshGitHub() {
+        commonSmartRefreshTest(SourceControl.GITHUB, "DockstoreTestUser2/hello-dockstore-workflow", "testBoth");
+    }
+
+    /**
+     * This tests that smart refresh correctly refreshes the right versions based on some scenarios for Gitlab
+     */
+    @Test
+    public void testSmartRefreshGitlab() {
+        commonSmartRefreshTest(SourceControl.GITLAB, "dockstore.test.user2/dockstore-workflow-example", "master");
+    }
+
+    /**
+     * This tests that smart refresh correctly refreshes the right versions based on some scenarios for BitBucket
+     */
+    @Test
+    public void testSmartRefreshBitbucket() {
+        commonSmartRefreshTest(SourceControl.BITBUCKET, "dockstore_testuser2/dockstore-workflow", "cwl_import");
+    }
+
+    private void commonSmartRefreshTest(SourceControl sourceControl, String workflowPath, String versionOfInterest) {
+        ApiClient client = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowsApi = new WorkflowsApi(client);
+
+        String correctDescriptorPath = "/Dockstore.cwl";
+        String incorrectDescriptorPath = "/Dockstore2.cwl";
+
+        String fullPath = sourceControl.toString() + "/" + workflowPath;
+
+        // Add workflow
+        workflowsApi.manualRegister(sourceControl.name(), workflowPath, correctDescriptorPath, "",
+                DescriptorLanguage.CWL.getLowerShortName(), "");
+
+        // Smart refresh individual that is valid (should add versions that doesn't exist)
```

Comment:
Comment seems reasonable, but this test doesn't seem to check this?

Reply:
Initially, the versions don't exist, so the initial refresh is testing this.